### PR TITLE
Refactor common domain transfer logic

### PR DESCRIPTION
**Pull request by wolfgangmeyers**

### CODE REVIEW

File: java/google/registry/flows/domain/DomainTransferRequestFlow.java [hunk 1]
```diff
@@ -155,6 +148,9 @@ public final EppResponse run() throws EppException {
             serverApproveNewExpirationTime,
             historyEntry,
             existingDomain,
+            targetId,
```
Comment (CydeWeys) [inline java/google/registry/flows/domain/DomainTransferRequestFlow.java]: This doesn't seem right to me?

Comment (wolfgangmeyers) [inline java/google/registry/flows/domain/DomainTransferRequestFlow.java]: targetId was previously used from a field. I can replace with existingDomain.getRepoId()

Comment (wolfgangmeyers) [inline java/google/registry/flows/domain/DomainTransferRequestFlow.java]: fixed

Comment (CydeWeys) [inline java/google/registry/flows/domain/DomainTransferRequestFlow.java]: Never mind, I thought this was creating a list, but it is just method parameters.

**Comment by CydeWeys** [comment]
Going through internal review now.

File: java/google/registry/flows/domain/DomainTransferUtils.java [hunk 2]
```diff
@@ -104,22 +103,29 @@ public static TransferData createPendingTransferData(
         .add(createTransferBillingEvent(
             automaticTransferTime,
             historyEntry,
-            targetId,
+            existingDomain.getRepoId(),
```
Comment (CydeWeys) [inline java/google/registry/flows/domain/DomainTransferUtils.java]: This change was wrong -- targetId is the fullyQualifiedDomainName, not the repoId.  I fixed it for you.

File: java/google/registry/flows/domain/DomainTransferUtils.java [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+import google.registry.model.transfer.TransferStatus;
+import javax.annotation.Nullable;
+import org.joda.money.Money;
+import org.joda.time.DateTime;
+import org.joda.time.Duration;
+
+/**
+ * Utility logic for facilitating domain transfers.
+ */
+public class DomainTransferUtils {
```
Comment (CydeWeys) [inline java/google/registry/flows/domain/DomainTransferUtils.java]: I made this final and added a private no-op constructor per our convention for static utility classes.

**Comment by CydeWeys** [comment]
Merged as b9222759dc89f6d1760aa07138b19db8cc927913.

**Pull request closed without merging**